pkg/objstore/azure/helpers.go

```diff
@@ -9,32 +9,30 @@ import (
 	blob "github.com/Azure/azure-storage-blob-go/azblob"
 )
 
-var (
-	blobFormatString = `https://%s.blob.core.windows.net`
-)
-
 // DirDelim is the delimiter used to model a directory structure in an object store bucket.
 const DirDelim = "/"
 
-func getContainerURL(ctx context.Context, accountName, accountKey, containerName string) (blob.ContainerURL, error) {
-	c, err := blob.NewSharedKeyCredential(accountName, accountKey)
+var errorCodeRegex = regexp.MustCompile(`X-Ms-Error-Code:\D*\[(\w+)\]`)
+
+func getContainerURL(conf Config) (blob.ContainerURL, error) {
+	c, err := blob.NewSharedKeyCredential(conf.StorageAccountName, conf.StorageAccountKey)
 	if err != nil {
 		return blob.ContainerURL{}, err
 	}
 	p := blob.NewPipeline(c, blob.PipelineOptions{
 		Telemetry: blob.TelemetryOptions{Value: "Thanos"},
```

Comment:
Add https://github.com/Azure/azure-storage-blob-go/blob/HEAD/azblob/zc_pipeline.go#L13

Timeout options from context

Reply:
Can this be another PR backed by an issue ? Not sure what values we should put in there. You already have defaults here: https://github.com/Azure/azure-storage-blob-go/blob/HEAD/azblob/zc_policy_retry.go#L70

Reply:
Not really as you removed context args everywhere.

ctx.Deadline() gives you time.Time & ok flag

You could just put that into you could put that into time.Now()- deadline into `TryTimeout` and IMO this would be good enough,

Reply:
@povilasv  Done ! Thanks for the feedback :smile: I learned something.